streampipes-client-python/streampipes_client/functions/broker/nats_broker.py

```diff
@@ -0,0 +1,81 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+from typing import AsyncIterator
+
+from nats import connect
+from nats.aio.client import Client
+from nats.aio.subscription import Subscription
+from streampipes_client.functions.broker.broker import Broker
+
+logger = logging.getLogger(__name__)
+
+
+class NatsBroker(Broker):
+    """Implementation of the NatsBroker"""
+
+    nats_client: Client
```

Comment:
This should be instance variables and not class variables right?

Reply:
That's right. Thanks for the detailed review